### DISCUSSION TITLE
Alternative to Authentication: Token header for Websocket in Flutter?

### DISCUSSION THREAD

**Discussion by peter-kenyon-theta** [deepgram/community#175] [General help] [answered]
Hi,

I am attempting to follow the text to speech tutorial here: https://blog.deepgram.com/flutter-speech-to-text-tutorial/
Unfortunately I don't currently have access to a device so I'm using an HTML target for my Flutter build.
It turns out that the IOWebsocketChannel class used by the tutorial is not supported for web clients, and the WebsocketChannel class which is fully cross-platform does not allow additional headers to be passed in the constructor. As a result I'm not able to send the Authorization header.

Is there an alternative method I could use to authenticate? e.g. can I pass an API token in the url or payload?

**Comment by nikolawhallon** [comment, marked as answer]
I am not a Flutter expert, but I know many exactly analogous situations - HTML/web builds indeed don't allow for the "usual" auth headers, but instead we support using the websocket `protocols` array, set to `["token", "your_api_key"]` - looking at the Flutter docs suggests this is supported:
```
WebSocketChannel.connect(Uri uri, {Iterable<String>? protocols})
```
(specifically this is setting the `Sec-WebSocket-Protocol` header). Maybe this points in the right direction - I can try with Flutter specifically if not, but this is a common recipe across web applications I think!
